cargo build --no-default-features failing

When trying to build without any features, it fails with a few errors:

```
error: A non-empty glob must import something with the glob's visibility
  --> src/cdef.rs:23:9
   |
23 | pub use self::native::*;
   |         ^^^^^^^^^^^^^^^

error[E0425]: cannot find value `CDEF_FILTER_FNS` in this scope
  --> src/asm/cdef.rs:76:13
   |
76 |       match CDEF_FILTER_FNS[cpu.as_index()][decimate_index(xdec, ydec)] {
   |             ^^^^^^^^^^^^^^^ not found in this scope
```

and some more that are similar. Looks like this was introduced with 79019d32fc101341ca840e0208a07fdb17b1023b. Seems like we need to add some more `cfg` checks to some parts of that code but I don't have enough knowledge of this code to do it myself.

Found it while building gst-plugins-rs that uses this.

--- a/.travis.yml
+++ b/.travis.yml
@@ -112,3 +112,9 @@ jobs:
         - cargo clippy -- -D warnings -A clippy::cast_lossless -A clippy::cast_ptr_alignment -A clippy::cognitive_complexity -A clippy::needless_range_loop -A clippy::too_many_arguments -A clippy::verbose_bit_mask -A clippy::unreadable_literal -A clippy::many_single_char_names --verbose
       env:
         - CACHE_NAME=CLIPPY
+    - name: "No default features"
+      rust: stable
+      script:
+        - cargo check --no-default-features
+      env:
+        - CACHE_NAME=CHECK

--- a/src/asm/mod.rs
+++ b/src/asm/mod.rs
@@ -7,5 +7,7 @@
 // Media Patent License 1.0 was not distributed with this source code in the
 // PATENTS file, you can obtain it at www.aomedia.org/license/patent.
 
+#[cfg(all(target_arch = "x86_64", feature = "nasm"))]
 pub mod cdef;
+#[cfg(all(target_arch = "x86_64", feature = "nasm"))]
 pub mod mc;

--- a/src/cdef.rs
+++ b/src/cdef.rs
@@ -20,7 +20,7 @@ use crate::cpu_features::CpuFeatureLevel;
 use std::cmp;
 
 #[cfg(any(not(target_arch = "x86_64"), not(feature = "nasm")))]
-pub use self::native::*;
+pub(crate) use self::native::*;
 #[cfg(all(target_arch = "x86_64", feature = "nasm"))]
 pub use crate::asm::cdef::*;
 

